synapse/rest/consent/consent_resource.py

```diff
@@ -142,10 +142,10 @@ def _async_render_GET(self, request):
         userhmac = None
         has_consented = False
         public_version = username == ""
-        if not public_version or not self.hs.config.user_consent_at_registration:
```

Comment:
Taking off the config flag causes a different set of errors. If the policy template doesn't correctly block usage of the missing variables, it will explode.

I'd highly recommend maintaining the flag.

Reply:
I don't follow I'm afraid. That if statement doesn't block calls to `_render_template`?

Reply:
Github helpfully wasn't showing me that `userhmac` is defined above here and therefore fine (I think). 

The problem is more of one where undefined variables can make it through `_render_template` and cause problems when the template tries to use them. When I was testing the template stuff originally, the thing exploded if stuff like `userhmac` wasn't defined at template time if the template didn't have an `!is_public` if statement around it.